src/OmniSharp.MSBuild/MSBuildProjectSystem.cs

```diff
@@ -483,6 +492,104 @@ private void UpdateReferences(Project project, IList<string> references)
             }
         }
 
+        private List<PackageDependency> CreatePackageDependencies(IEnumerable<PackageReference> packageReferences)
+        {
+            var list = new List<PackageDependency>();
+
+            foreach (var packageReference in packageReferences)
+            {
+                var dependency = new PackageDependency
+                {
+                    Name = packageReference.Identity.Id,
+                    Version = packageReference.Identity.Version?.ToNormalizedString()
+                };
+
+                list.Add(dependency);
+            }
+
+            return list;
+        }
+
+        private void CheckForUnresolvedDependences(ProjectFileInfo projectFileInfo, ProjectFileInfo previousProjectFileInfo = null, bool allowAutoRestore = false)
+        {
+            List<PackageDependency> unresolvedDependencies;
+
+            if (!File.Exists(projectFileInfo.ProjectAssetsFile))
+            {
+                // Simplest case: If there's no lock file and the project file has package references,
+                // there are certainly unresolved dependencies.
+                unresolvedDependencies = CreatePackageDependencies(projectFileInfo.PackageReferences);
+            }
+            else
+            {
+                // Note: This is a bit of misnmomer. It's entirely possible that a package reference has been removed
+                // and a restore needs to happen in order to update project.assets.json file. Otherwise, the MSBuild targets
+                // will still resolve the removed reference as a reference in the user's project. In that case, the package
+                // reference isn't so much "unresolved" as "incorrectly resolved".
+                IEnumerable<PackageReference> unresolvedPackageReferences;
+
+                // Did the project file change? Diff the package references and see if there are unresolved dependencies.
+                if (previousProjectFileInfo != null)
```

Comment:
would be really good if NuGet.ProjectModel provided some API for these types of checks